Automattic/woocommerce-payments
# Fix capture payments later

Fixes #2623 

#### Changes proposed in this Pull Request





This PR adds an implementation to pass on the capture method when creating a payment intent. On UPE, a payment intention is created when checkout loads and it was not including the capture method information that is saved in the settings. Because of that, all payments were being automatically captured.

The present changes fixes the manual capture issue on payments with new credit cards. It basically checks the settings and send that information to the Payments API.

If you have card + other payment methods enabled, you might notice that, when you have manual capture enabled, you will only see card as an option on checkout. The reason for that is simply because we're deliberately limiting the payment methods to card, as manual capture only works with it. There is an ongoing discussion on how to properly communicate that for the merchants and you can find more info on p1631843780047400-slack-CHG7MTCAF 





#### Testing instructions




- Visit [the store](http://localhost:8082/) and log in as admin.
- In [WooCommerce Payments settings](http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments), make sure `Issue an authorization on checkout, and capture later` option is unticked and saved.
- Go to the store > `Shop` and add a product to the cart.
- Go to checkout and pay using a new test card (e.g. 4242 4242 4242 4242).
- In the [Orders](http://localhost:8082/wp-admin/edit.php?post_type=shop_order), you should see the order you just placed with status `Processing`, as usual.
- Go to [WooCommerce Payments settings](http://localhost:8082/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments) again, tick `Issue an authorization on checkout, and capture later` and save.
- Place a new order following the steps above.
- In the [Orders](http://localhost:8082/wp-admin/edit.php?post_type=shop_order), you should see the order you just placed with status `On hold` and a similar order note:
>A payment of €xx.xx was authorized using WooCommerce Payments (pi_...). Order status changed from Pending payment to On hold.

-------------------


**Post merge**
